### PR TITLE
[8.0][IMP] stock_quant_partner_info: some changes in original module (avanzosc/odoo-addons) ported

### PR DESCRIPTION
**Pull request by oihane**
* remove location usage filter
* force move order to take last movement

### CODE REVIEW

**Comment by oihane** [comment]
Comparing the module merged and the one our customers are using I found that there were some changes that I forgot to migrate to the module in the PR. Can you please check them out?
cc @Cedric-Pigeon @pedrobaeza @andhit-r

**Review by agaldona — APPROVED** [review]

**Comment by oihane** [comment]
Please @Cedric-Pigeon @pedrobaeza @andhit-r can you check it out, there is a bug that makes it fail.

```
  File "/OCA/stock-logistics-warehouse/stock_quant_partner_info/models/stock_quant.py", line 21, in _compute_partner_id
    move.picking_id.partner_id.commercial_partner_id
  File "/core/openerp/fields.py", line 825, in __get__
    record.ensure_one()
  File "/core/openerp/models.py", line 5345, in ensure_one
    raise except_orm("ValueError", "Expected singleton: %s" % self)
except_orm: ('ValueError', 'Expected singleton: stock.move(254, 205, 177, 188)')
```

**Review by andhit-r — APPROVED** [review]

**Review by gurneyalex — APPROVED** [review]

File: stock_quant_partner_info/models/stock_quant.py [hunk 1]
```diff
@@ -14,8 +14,9 @@ class StockQuant(models.Model):
 
     @api.depends('location_id', 'history_ids')
     def _compute_partner_id(self):
-        for quant in self.filtered(
-                lambda q: q.location_id.usage in ['customer', 'supplier']):
-            move = quant.history_ids[:-1]
+        for quant in self:
+            move = quant.history_ids.sorted(key=lambda m: m.date)[-1:]
```
Comment (gurneyalex) [inline stock_quant_partner_info/models/stock_quant.py:18]: you can get > 1 move here, and this will crash on line 20 below as reported byt someone else.

**Review by gurneyalex — CHANGES_REQUESTED** [review]

File: stock_quant_partner_info/models/stock_quant.py [hunk 2]
```diff
@@ -14,8 +14,9 @@ class StockQuant(models.Model):
...
             quant.dest_partner_id =\
-                move.picking_id.partner_id.commercial_partner_id
+                move.picking_id.partner_id.commercial_partner_id\
+                if move.location_id.usage not in ['customer', 'supplier'] else\
```
Comment (gurneyalex) [inline stock_quant_partner_info/models/stock_quant.py:21]: you are inverting the test from the original code here

**Comment by gurneyalex** [comment]
overridden in #394

File: stock_quant_partner_info/models/stock_quant.py (hunk 1 above)
Comment (pedrobaeza) [reply stock_quant_partner_info/models/stock_quant.py:18]: Not in this case. With this you always get one record. If you make `[:-1]` it's when you get more records.

Comment (gurneyalex) [reply stock_quant_partner_info/models/stock_quant.py:18]: @pedrobaeza What I am fixing is https://github.com/OCA/stock-logistics-warehouse/pull/291#issuecomment-311901848

**Pull request closed without merging**